CliFx/Domain/CommandArgumentSchema.cs

```diff
@@ -18,6 +18,37 @@ internal abstract partial class CommandArgumentSchema
 
         public bool IsScalar => TryGetEnumerableArgumentUnderlyingType() == null;
 
+        public IEnumerable<string> GetValidValues()
```

Comment:
Let's change the return type to `IReadOnlyList<string>` since we want it buffered and we'll be enumerating it multiple times.

Reply:
Moved this logic a new method `IEnumerable<string> EnumerateValidValues()` and refactoring `GetValidValues()` to return `IReadOnlyList<string>` by lazily caching the result of `EnumerateValidValues()` in a private field.